src/main/integrations/sentry-minidump/index.ts

```diff
@@ -195,56 +200,58 @@ export class SentryMinidump implements Integration {
     //     about it. Just use the breadcrumbs and context information we have
     //     right now and hope that the delay was not too long.
 
-    const uploader = this._uploader;
-    if (uploader === undefined) {
+    if (this._minidumpLoader === undefined) {
       throw new SentryError('Invariant violation: Native crashes not enabled');
     }
 
     try {
-      const paths = await uploader.getNewMinidumps();
+      const minidumps = await this._minidumpLoader();
 
-      if (paths.length > 0) {
+      if (minidumps.length > 0) {
         const hub = getCurrentHub();
-        const enabled = hub.getClient()?.getOptions().enabled;
+        const client = hub.getClient();
+
+        if (!client) {
+          return true;
+        }
+
+        const enabled = client.getOptions().enabled;
 
         // If the SDK is not enabled, we delete the minidump files so they
         // dont accumulate and/or get sent later
         if (enabled === false) {
-          paths.forEach((path) => forget(unlinkAsync(path)));
+          minidumps.forEach(deleteMinidump);
           return false;
         }
 
-        const currentCloned = Scope.clone(hub.getScope());
         const storedScope = Scope.clone(this._scopeLastRun);
         let newEvent = await storedScope.applyToEvent(event);
 
-        if (newEvent) {
-          newEvent = await currentCloned.applyToEvent(newEvent);
-
-          // We handle beforeSend and sampleRate manually here because native crashes do not go through @sentry/core
-          const { beforeSend, sampleRate } = options;
-          if (typeof sampleRate === 'number' && Math.random() > sampleRate) {
```

Comment:
Why did we get rid of this logic here?

Reply:
`beforeSend` and `sampleRate` handling now all occur in `@sentry/core`. 

This only existed before because native crashes were not sent through the underlying JavaScript SDK and were instead hacked on top of the transport.